contracts/clearing/BrokerVault.sol

```diff
@@ -12,24 +13,39 @@ contract BrokerVault is IVault2, ISettle, Ownable2Step, ReentrancyGuard {
 	/// that may not return a boolean value on success.
 	using SafeERC20 for IERC20;
 
+	// ====== Variables ======
+
+	address public broker;
+	TradingApp public tradingApp;
+	mapping(bytes32 channelId => bool done) public performedSettlements;
+
 	mapping(address token => uint256 balance) internal _balances;
-	TradingApp internal tradingApp;
-	mapping(uint256 channel_id => bool done) internal performedSettlements;
-	address public _broker;
+
+	// ====== Errors ======
+
+	error InsufficientBalance(address token, uint256 required, uint256 available);
+	error InvalidAddress();
+	error InvalidAmount(uint256 amount);
+	error SettlementAlreadyPerformed(bytes32 channelId);
+	error BrokerNotParticipant(address actual, address expectedBroker);
 
 	// ====== Constructor ======
 
-	constructor(address owner, address broker_) Ownable(owner) {
+	constructor(address owner, address broker_, TradingApp tradingApp_) Ownable(owner) {
 		broker = broker_;
+		tradingApp = tradingApp_;
 	}
 
 	// ---------- View functions ----------
 
-	function balanceOf(address token) external view returns (uint256) {
+	function balanceOf(address user, address token) external view returns (uint256) {
 		return _balances[token];
```

Comment:
For better compatibility and to follow PoLA, let's return 0 if user != broker.

Reply:
cb9be30